### DISCUSSION TITLE
How To Create User After Provider Sign In

### DISCUSSION THREAD

**Discussion by masnwilliams** [supabase/supabase#7213] [Questions] [answered]
So I am trying to add the ability to "Sign Up" with OAuth. Right now all it does is sign the user in and create an auth user if they don't already have one but I also want to create a user in my public schema the first time they use OAuth to sign in.

I'm not sure if there is an ideal way to go about doing this but any help is appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
Normally you would use a trigger on auth.users insert and insert your desired data in the public table.  Depending on steps involved you might also want that trigger to handle updates. 
https://supabase.com/docs/guides/auth/managing-user-data#advanced-techniques

**Reply by masnwilliams** [reply]
I had some issues with triggers in the past that I think were related to using Prisma ORM. I'll give it another shot though. Thanks!

**Reply by masnwilliams** [reply]
Got it working! Thank you so much!
